src/vulkan-renderer/application.cpp

```diff
@@ -48,6 +49,10 @@ void Application::mouse_button_callback(GLFWwindow *window, int button, int acti
     }
 }
 
+void Application::mouse_scroll_callback(GLFWwindow *window, double xoffset, double yoffset) {
+    m_camera->process_mouse_scroll(static_cast<float>(yoffset));
```

Comment:
Isn't that already game code? Iam still not sure where we split up game/ engine code.

Reply:
Yes, that's actually tightly coupled.

Engine:
1. Provide a camera system
2. Provide a key and mouse input system
3. Make both available via an API
4. Do not bind keys / mouse / buttons with any specific functionality

Game:
1. Use the API in order to couple keys/mouse with camera(s)

But for now there is no game, so maybe it's enough to add a TODO?

Reply:
> But for now there is no game, so maybe it's enough to add a TODO?
 
 We have the example folder, build for it actually.

Reply:
Yes we need further separation between library and the rest.

Reply:
The entire `Application` is already part of the game tbh.

Reply:
Let's continue discussion here: https://github.com/inexorgame/vulkan-renderer/issues/297.